src/lib/utils/meta_tables/abstract_meta_system_table.cpp

```diff
@@ -0,0 +1,46 @@
+#ifdef __APPLE__
+#include <mach/mach.h>
+#endif
+
+#include <fstream>
+
+#include "utils/meta_tables/abstract_meta_system_table.hpp"
+
+namespace opossum {
+
+AbstractMetaSystemTable::AbstractMetaSystemTable(const TableColumnDefinitions& column_definitions)
+    : AbstractMetaTable(column_definitions) {}
+
+// Returns the number of logical processors
+size_t AbstractMetaSystemTable::_get_cpu_count() {

```

Comment:
Too little to gain by changing this now, so just as a random thought: The two AbstractMetaSystemTables don't have any observable behavior in common. Building a class hierarchy just to access `_get_cpu_count` feels like overkill, if the method could also have been a helper method somewhere.

Reply:
where would we preferably place such a helper method?

Reply:
Wait a second. How about simply using `Topology::num_cpus`?

Reply:
Correction: num_cpus returns the number of used CPUs. Still, I'd consider exposing numa_num_configured_cpus through the Topology class.

Reply:
how come that noone ever mentioned `Topology` until now? 
I think we could drop the whole AbstractMetaSystemTable in favor of Topology::num_cpus.
This number is only needed to normalize the runtimes, so we actually want to know the numbers of used cpus and not potentially useable cpus.

Reply:
I don't think that `Topology` was touched in the recent past. Only the tales of the elders of Hyrise remind of this mystical class and those elders are sometimes doddery :/

Reply:
Sorry, haven't used the Topology in years.

Reply:
Another thought: The current approach might be a problem if the scheduler is used. As the scheduler spawns worker threads with a certain CPU affinity, `_get_cpu_count` might return the affinity of the worker, not that of the entire process.

Not sure if there is a nice way to automatically test this, but we should at least have a look if the values make sense.

Reply:
this seems to be a real problem. to solve this, i now return topology.num_cpus as cpu_affinity_count. this is true since the number of cpus in the topology is set according to the available cpus. however, if we change the cpu affinities without reinitializing the topology, this won't reflect in the metatable.

Reply:
> however, if we change the cpu affinities without reinitializing the topology, this won't reflect in the metatable.

Shouldn't do that anyway ;)